websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/validator/ServiceDeclarationValidator.java

```diff
@@ -0,0 +1,252 @@
+                    .map(param -> Arrays.stream(param.trim().split("\\|"))
+                            .map(String::trim).collect(Collectors.toList())
+                    ).collect(Collectors.toList());
+            SeparatedNodeList<ParameterNode> availableParameters = functionDefinition.functionSignature().parameters();
+            if (availableParameters.size() >= 1) {
+                availableParameters
+                        .stream()
+                        .map(param -> (RequiredParameterNode) param)
+                        .filter(param -> isParamNotAllowed(allowedParams, param))
+                        .forEach(param -> {
+                            String paramType = param.typeName().toString();
+                            WebSubDiagnosticCodes errorCode = WebSubDiagnosticCodes.WEBSUB_105;
+                            updateContext(
+                                    context, errorCode, functionDefinition.location(), paramType.trim(), functionName);
+                        });
+            } else {
+                if (!allowedParameters.isEmpty()) {
+                    WebSubDiagnosticCodes errorCode = WebSubDiagnosticCodes.WEBSUB_106;
+                    updateContext(
+                            context, errorCode, functionDefinition.location(), functionName,
+                            String.join(",", allowedParameters));
+                }
+            }
+        }
+    }
+
+    private boolean isParamNotAllowed(List<List<String>> allowedParams, RequiredParameterNode param) {
+        String paramType = param.typeName().toString();
+        return Arrays
+                .stream(paramType.trim().split("\\|"))
+                .map(String::trim)
+                .anyMatch(paramsTypes ->
+                        allowedParams.stream().noneMatch(p -> p.contains(paramsTypes))
+                );
+    }
+
+    private void validateMethodReturnTypes(SyntaxNodeAnalysisContext context,
+                                           FunctionDefinitionNode functionDefinition) {
+        String functionName = functionDefinition.functionName().toString();
+        List<String> predefinedReturnTypes = allowedReturnTypes.get(functionName);
+        if (allowedMethods.contains(functionName)) {
+            Optional<ReturnTypeDescriptorNode> returnTypesOpt = functionDefinition.functionSignature().returnTypeDesc();
+            if (returnTypesOpt.isPresent()) {
+                ReturnTypeDescriptorNode returnTypeDescription = returnTypesOpt.get();
+                Node returnType = returnTypeDescription.type();
+                String returnTypeName = returnType.toString().trim();
+                List<String> availableReturnTypes = Arrays
+                        .stream(returnTypeName.replace(Constants.OPTIONAL, "").split("\\|"))
+                        .map(String::trim)
+                        .collect(Collectors.toList());
+                if (!predefinedReturnTypes.containsAll(availableReturnTypes)) {
```

Comment:
Rather doing a string match for return types, it's better of you use the sematic model and do proper type check. 
```
websub:Acknowledgement
```
can be stated with an alias
```
import ballerina/websub as hello;
hello:Acknowledgement
```

Reply:
`context.semanticModel().symbol(member).get()`